client/src/components/App.jsx

```diff
@@ -24,11 +39,90 @@ class App extends React.Component {
       });
   }
 
+  /* componentWillUnmount() {
+    document.removeEventListener('mousedown', this.handleClickOutside);
+  } */
+
+  setWrapperRef(node) {
+    console.log('inside wrapper');
+    this.wrapperRef = node;
+    console.log(node);
+  }
+
+  handleOnChange(e) {
+    const stateObj = this.state;
+    if (stateObj.isVisible) {
+      this.setState({ isVisible: true });
+    }
+
+    const userId = 1;
+    const categoryId = e.target.value;
+    fetch(`filter/${userId}/${categoryId}`)
+      .then((response) => response.json())
+      .then((data) => {
+        this.setState({
+          username: data.user.username,
+          videos: data.videos,
+          currentCategory: data.videos[0].category.name,
+        });
+      });
+  }
+
+  handleOnRightClick() {
+    document.getElementById('container').scrollLeft += 50;
+  }
+
+  handleOnLeftClick() {
+    document.getElementById('container').scrollLeft -= 50;
+  }
+
+  handleOnClickMenu() {
+    const stateObj = this.state;
+    const isVisible = !stateObj.isVisible;
+    this.setState({ isVisible });
```

Comment:
Or...
const { isVisible } = this.state;  // the whole state isn't needed, just one thing
this.setState({ isVisible: !isVisible })